kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/balance/presenter/BalancePresenter.java

```diff
@@ -36,22 +90,159 @@ private void updateBalance(Balance balance) {
 		if (balanceValue == 0) {
 			balanceString = BALANCE_ZERO_TEXT;
 		} else {
-			balanceString = StringUtil.getAmountFormatted(balanceValue);
+			balanceString = getAmountFormatted(balanceValue);
 		}
 		if (view != null) {
 			view.updateBalance(balanceString);
 		}
 	}
 
+	private int getType(OfferType offerType) {
+		switch (offerType) {
+			case SPEND:
+				return SPEND;
+			default:
+			case EARN:
+				return EARN;
+		}
+	}
+
+	private void createOrderObserver() {
+		orderObserver = new Observer<Order>() {
+			@Override
+			public void onChanged(Order order) {
+				if (order.getOrigin() == Origin.MARKETPLACE) {
+					switch (order.getStatus()) {
+						case PENDING:
+							incrementPendingCount();
+							currentPendingOrder = order;
+							status = getStatus(order);
+							updateSubTitle(order.getAmount(), status, getType(order.getOfferType()));
+							break;
+						case COMPLETED:
+						case FAILED:
+							decrementPendingCount();
+						case DELAYED:
+							if (isCurrentOrder(order)) {
+								status = getStatus(order);
+								updateSubTitle(order.getAmount(), status, getType(order.getOfferType()));
+							}
+							break;
+					}
+				}
+
+			}
+		};
+	}
+
+	private void incrementPendingCount() {
+		pendingOrderCount++;
+	}
+
+	private void decrementPendingCount() {
+		if(pendingOrderCount > 0) {
+			pendingOrderCount--;
+		}
+	}
+
+	private boolean isCurrentOrder(Order order) {
+		return order != null && currentPendingOrder != null && currentPendingOrder.equals(order);
+	}
+
+	private @OrderStatus int getStatus(Order order) {
+		switch (order.getStatus()) {
+			case COMPLETED:
+				return COMPLETED;
+			case FAILED:
+				return FAILED;
+			case DELAYED:
+				return DELAYED;
+			default:
+			case PENDING:
+				return PENDING;
+		}
+	}
+
+
+	private void updateSubTitle(int amount, @OrderStatus int status, @OrderType int offerType) {
+		if (view != null) {
+			view.updateSubTitle(amount, status, offerType);
+		}
+	}
+
 	@Override
 	public void onAttach(IBalanceView view) {
 		super.onAttach(view);
+		showWelcomeToKin();
+		addObservers();
+	}
+
+	private void showWelcomeToKin() {
+		if (view != null) {
+			view.setWelcomeSubtitle();
+		}
+	}
+
+	private void addObservers() {
+		orderRepository.addOrderObserver(orderObserver);
 		blockchainSource.addBalanceObserver(balanceObserver);
 	}
 
 	@Override
 	public void onDetach() {
 		super.onDetach();
+		removeObservers();
+	}
+
+	private void removeObservers() {
+		orderRepository.removeOrderObserver(orderObserver);
 		blockchainSource.removeBalanceObserver(balanceObserver);
 	}
+
+	@Override
+	public void balanceClicked() {
+		eventLogger.send(BalanceTapped.create());
+		if (this.balanceClickListener != null) {
+			this.balanceClickListener.onClick();
```

Comment:
see my other related comment - the back and forth seems to be redundant

Reply:
it's not, see my comments above.

Reply:
ok you are right:)